raiden-ts/src/path/epics.ts

```diff
@@ -0,0 +1,166 @@
+                  filter(
+                    partner =>
+                      channelCanRoute(
+                        state,
+                        presences,
+                        tokenNetwork,
+                        partner,
+                        action.meta.value,
+                      ) === true,
+                  ),
+                  // TODO: request pathFind once from ourselves instead of each partner
+                  // this requires us updating the PFS with our outgoing capacity on each channel
+                  mergeMap(partner =>
+                    fromFetch(`${pfs}/api/v1/${tokenNetwork}/paths`, {
+                      method: 'POST',
+                      headers: { 'Content-Type': 'application/json' },
+                      body: losslessStringify({
+                        from: partner,
+                        to: target,
+                        value: UInt(32).encode(action.meta.value),
+                        max_paths: 10, // eslint-disable-line @typescript-eslint/camelcase
+                      }),
+                    }).pipe(
+                      timeout(httpTimeout),
+                      mergeMap(async response => {
+                        if (!response.ok)
+                          throw new Error(
+                            `PFS: paths request: code=${
+                              response.status
+                            } => body=${await response.text()}`,
+                          );
+                        const decoded = PathResults.decode(await response.json());
+                        if (isLeft(decoded)) throw ThrowReporter.report(decoded);
+                        return decoded.right;
+                      }),
+                      // catch early so other output routes can have the change to succeed
+                      catchError(err => {
+                        console.error('PFS: request error - ignoring', err);
+                        return of<PathResults>({ result: [] });
+                      }),
+                    ),
+                  ),
+                  reduce<PathResults, PathResults['result']>(
+                    (acc, { result }) => [...acc, ...result],
+                    [],
+                  ),
+                  map(result => ({
+                    routes: result
+                      .slice()
+                      .sort(({ estimated_fee: a }, { estimated_fee: b }) => a - b)
+                      .map(({ path }) => ({ route: path })),
+                  })),
+                );
+              } else {
+                throw new Error(`PFS disabled and no direct route available`);
+              }
+            }),
+            withLatestFrom(statePresencesConfig$),
+            // validate/cleanup received routes/metadata
+            map(([{ routes: result }, [state, presences]]) => {
+              const routes: { route: readonly Address[] }[] = [],
```

Comment:
Should we already include the `estimated_fee` here or will we have a new epic that handles estimation requests?

Reply:
It'll be here, but I expected to expose it only when we work on supporting the paid PFS.